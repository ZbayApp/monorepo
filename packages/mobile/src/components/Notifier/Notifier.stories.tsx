import React from 'react'
import { storiesOf } from '@storybook/react-native'

import { appImages } from '../../assets'
import { storybookLog } from '../../utils/functions/storybookLog/storybookLog.function'

import { Notifier } from './Notifier.component'

storiesOf('Notifier', module).add('Default', () => (
  <Notifier
    onButtonPress={storybookLog('button pressed')}
    onEmailPress={storybookLog('email pressed')}
    icon={appImages.update_graphics}
    title={'Coming update will remove communities & messages'}
    message={
      "Quiet's next release includes major changes to authentication and encryption that will let us offer DMs, private channels, roles, multi-device support, and user removal in future versions! ðŸŽ‰ However, these changes are not backwards compatible, so you must re-install Quiet from tryquiet.org and re-create or re-join your community. ðŸ˜¥ This version of Quiet will no longer receive any updates or security fixes, so please re-install soon. We apologize for the inconvenience."
    }
  />
))
