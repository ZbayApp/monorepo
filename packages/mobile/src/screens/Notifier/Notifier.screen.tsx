import React, { FC, useCallback } from 'react'
import { Linking } from 'react-native'
import { Notifier } from '../../components/Notifier/Notifier.component'
import { appImages } from '../../assets'
import { useDispatch } from 'react-redux'
import { navigationActions } from '../../store/navigation/navigation.slice'

export const NotifierScreen: FC = () => {
  const dispatch = useDispatch()

  const redirection = useCallback(() => {
    dispatch(navigationActions.pop())
  }, [dispatch])

  const helpline = useCallback(async () => {
    const link = 'mailto:help@quiet.chat'
    await Linking.openURL(link)
  }, [])

  return (
    <Notifier
      onButtonPress={redirection}
      onEmailPress={helpline}
      icon={appImages.update_graphics}
      title={'Coming update will remove communities & messages'}
      message={
        "Quiet's next release includes major changes to authentication and encryption that will let us offer DMs, private channels, roles, multi-device support, and user removal in future versions! ðŸŽ‰ However, these changes are not backwards compatible, so you must re-install Quiet from tryquiet.org and re-create or re-join your community. ðŸ˜¥ This version of Quiet will no longer receive any updates or security fixes, so please re-install soon. We apologize for the inconvenience."
      }
    />
  )
}
