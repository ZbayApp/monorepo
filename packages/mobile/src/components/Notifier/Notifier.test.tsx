import React from 'react'
import '@testing-library/jest-native/extend-expect'
import { fireEvent, screen } from '@testing-library/react-native'

import { renderComponent } from '../../utils/functions/renderComponent/renderComponent'

import { appImages } from '../../assets'

import { Notifier } from './Notifier.component'

describe('Notifier component', () => {
  it('should match inline snapshot', () => {
    const { toJSON } = renderComponent(
      <Notifier
        onButtonPress={jest.fn()}
        onEmailPress={jest.fn()}
        icon={appImages.update_graphics}
        title={'Coming update will remove communities & messages'}
        message={
          "Quiet's next release includes major changes to authentication and encryption that will let us offer DMs, private channels, roles, multi-device support, and user removal in future versions! ðŸŽ‰ However, these changes are not backwards compatible, so you must re-install Quiet from tryquiet.org and re-create or re-join your community. ðŸ˜¥ This version of Quiet will no longer receive any updates or security fixes, so please re-install soon. We apologize for the inconvenience."
        }
      />
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should respond on button tap', () => {
    const buttonCallback = jest.fn()

    renderComponent(
      <Notifier
        onButtonPress={buttonCallback}
        onEmailPress={jest.fn()}
        icon={appImages.update_graphics}
        title={'Coming update will remove communities & messages'}
        message={
          "Quiet's next release includes major changes to authentication and encryption that will let us offer DMs, private channels, roles, multi-device support, and user removal in future versions! ðŸŽ‰ However, these changes are not backwards compatible, so you must re-install Quiet from tryquiet.org and re-create or re-join your community. ðŸ˜¥ This version of Quiet will no longer receive any updates or security fixes, so please re-install soon. We apologize for the inconvenience."
        }
      />
    )

    fireEvent.press(screen.getByText('I understand'))

    expect(buttonCallback).toBeCalled()
  })
})
