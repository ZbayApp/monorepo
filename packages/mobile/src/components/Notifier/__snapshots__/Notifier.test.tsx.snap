// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifier component should match inline snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../assets/icons/update_graphics.png",
      }
    }
    style={
      {
        "height": 135,
        "margin": 20,
        "resizeMode": "cover",
        "width": 375,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "column",
        "gap": 20,
        "maxWidth": 330,
      }
    }
  >
    <Text
      color="main"
      fontSize={28}
      fontWeight="medium"
      horizontalTextAlign="center"
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Rubik-Medium",
            "fontSize": 28,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
        ]
      }
      verticalTextAlign="center"
    >
      Coming update will remove communities & messages
    </Text>
    <Text
      color="main"
      fontSize={14}
      fontWeight="normal"
      horizontalTextAlign="center"
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Rubik-Regular",
            "fontSize": 14,
            "textAlign": "center",
            "textAlignVertical": "center",
          },
        ]
      }
      verticalTextAlign="center"
    >
      "Quiet's next release includes major changes to authentication and encryption that will let us offer DMs, private channels, roles, multi-device support, and user removal in future versions! ðŸŽ‰ However, these changes are not backwards compatible, so you must re-install Quiet from tryquiet.org and re-create or re-join your community. ðŸ˜¥ This version of Quiet will no longer receive any updates or security fixes, so please re-install soon. We apologize for the inconvenience."
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#521C74",
          "borderRadius": 8,
          "justifyContent": "center",
          "marginVertical": 12,
          "minHeight": 50,
          "paddingVertical": 12,
          "width": 135,
        }
      }
      testID="button"
    >
      <Text
        color="white"
        fontSize={16}
        horizontalTextAlign="left"
        style={
          [
            {
              "color": "#ffffff",
              "fontFamily": "Rubik-Regular",
              "fontSize": 16,
              "textAlign": "left",
              "textAlignVertical": "center",
            },
          ]
        }
        verticalTextAlign="center"
      >
        I understand
      </Text>
    </View>
    <Text
      color="gray50"
      fontSize={16}
      horizontalTextAlign="left"
      onPress={[MockFunction]}
      style={
        [
          {
            "color": "#7F7F7F",
            "fontFamily": "Rubik-Regular",
            "fontSize": 16,
            "textAlign": "left",
            "textAlignVertical": "center",
          },
        ]
      }
      verticalTextAlign="center"
    >
      Need help? 
      <Text
        style={
          {
            "textDecorationLine": "underline",
          }
        }
      >
        help@quiet.chat
      </Text>
    </Text>
  </View>
</View>
`;
